src/box/index.cc

```diff
@@ -449,21 +451,61 @@ box_index_iterator(uint32_t space_id, uint32_t index_id, int type,
 	uint32_t part_count = mp_decode_array(&key);
 	if (key_validate(index->def, itype, key, part_count))
 		return NULL;
+	const char *pos, *pos_end;
+	char *pos_buf = NULL;
+	uint32_t pos_buf_size = 0;
+	auto pos_guard = make_scoped_guard([&pos_buf, &pos_buf_size] {
```

Comment:
Nit: Why do use capture by reference (&) here? Capture by value should do - these are plain pointers. Also, we usually use "capture everything by value" in guards: `make_scoped_guard([=])`.

Reply:
> But it does not work here - `pos_buf` is `NULL` initially

Right, missed that. Looks confusing, but I don't see how we could do better. Let's leave as is.